cloudconfig.go

```diff
@@ -0,0 +1,71 @@
+package cloudconfig
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"html/template"
+	"io/ioutil"
+
+	"github.com/giantswarm/awstpr"
+	tpraws "github.com/giantswarm/awstpr/aws"
+	"github.com/giantswarm/clustertpr/node"
+)
+
+type AWSNode struct {
+	Node    node.Node
+	AwsInfo tpraws.Node
+}
+
+type CloudConfigTemplateParams struct {
+	Spec      awstpr.Spec
+	Node      AWSNode
+	TLSAssets CompactTLSAssets
+}
+
+type CloudConfig struct {
+	config   string
+	params   CloudConfigTemplateParams
+	path     string
+	template string
+}
+
+func NewCloudConfig(templatePath string, params CloudConfigTemplateParams) (*CloudConfig, error) {
+	cloudConfigTempl, err := ioutil.ReadFile(templatePath)
```

Comment:
Sorry if I will sound a little bit harsh, but TBH, I don't care very much - I see nothing wrong in having a template file and just reading it. I also don't get the argument about "having hard runtime dependencies", since reading the file which is in the repo brings no runtime dependencies - even on OS X if that's what you meant.

I think that having the content of template in go code isn't a very good idea since it's very large and we will lose benefits of having a separate yaml file - first of them is that text editors will not provide yaml-specific colors or features if we put this in Go code.

If that will make you happier, I can move reading file out from this function and use afero, but as I wrote above, I'm struggling to see any point of that.